### PR TITLE
Feat: CRUD frontend connection

### PR DESCRIPTION
**Pull request by maxlshk**
- Implementation of CRUD operations on frontend, connected to backend. 
- Adaptability for all kinds of displays using `useMediaQuery` hook.
- Implementation of basic routes, no private routes yet.
- Inputs and forms separated into shared components. 
- Basic styling with Emotion.css.
- Added CORS on backend.

### CODE REVIEW

File: packages/frontend/src/modules/app/components/TodoItem/TodoItem.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	const handleComplete = (): void => {
+		const { id, ...rest } = todo;
+		updateTodo(id, { ...rest, completed: !todo.completed });
+	};
+
+	const handleDelete = (): Promise<void> => deleteTodo(todo.id);
+
+	const handleEdit = (): void => {
+		navigator(`/todos/edit/${todo.id}`);
```
Comment (malyshkoserhii) [inline packages/frontend/src/modules/app/components/TodoItem/TodoItem.tsx:25]: ROUTER_KEYS

File: packages/frontend/src/modules/app/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	if (todos.length === 0) {
+		return (
+			<div className={emptyStateStyles}>
+				No todos found. Add a new one!
+			</div>
+		);
+	}
+	const filteredTodos = todos.filter((todo) => {
+		if (filter === 'ALL') {
```
Comment (malyshkoserhii) [inline packages/frontend/src/modules/app/components/TodoList/TodoList.tsx:35]: use enum 'ALL', 'PRIVATE' etc.

replace lots of if tatements with object with dynamic keys

File: packages/frontend/src/modules/app/pages/EditTodoPage/EditTodoPage.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	const { handleSubmit, reset, register } = useForm<FormData>();
+	const { todos, updateTodo } = useTodoStore();
+
+	const todo = todos.find((todo) => todo.id === Number(id));
+
+	const onSubmit = (data: FormData): void => {
+		console.log('edit data', data);
+		updateTodo(Number(id), data);
+		reset();
+		navigate('/todos');
```
Comment (malyshkoserhii) [inline packages/frontend/src/modules/app/pages/EditTodoPage/EditTodoPage.tsx:22]: ROUTER_KEYS

File: packages/frontend/src/modules/app/pages/EditTodoPage/EditTodoPage.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+			title={'Create new todo'}
+		>
+			<TextInput
+				name="title"
+				register={register}
+				placeholder="Title"
+				defaultValue={todo.title}
+				type="input"
+				required
+			/>
```
Comment (malyshkoserhii) [inline packages/frontend/src/modules/app/pages/EditTodoPage/EditTodoPage.tsx:38]: add validation on FE side

File: packages/frontend/src/shared/services/http.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import axios from 'axios';
+import { Todo } from '../services/types';
+
+const API_URL = 'http://localhost:3030/api';
```
Comment (malyshkoserhii) [inline packages/frontend/src/shared/services/http.ts:4]: .env

File: packages/frontend/src/shared/services/http.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+export const getTodos = async (): Promise<Todo[]> => {
```
Comment (malyshkoserhii) [inline packages/frontend/src/shared/services/http.ts:6]: create HttpService. Please see company codestyle

**Comment by maxlshk** [comment]
Fix: all pointed out issues reviewed and fixed ✓

- Used ROUTER_KEYS instead of navigation routes as strings.
- Replaced if-statements by using enum functionality.
- Added validation for form inputs.
- Moved necessary variables to .env file.
- Created `http.service.ts` as generic service class for CRUD operations and `todo.service.ts` file extending it's functionality.

Additionally:

- Added todo creation/editing form submission error status to handle potential unsuccessful operations.

File: packages/frontend/src/shared/components/textinput/textinput.component.tsx [hunk 7]
```diff
@@ -21,17 +28,41 @@ export const TextInput: React.FC<FormFieldProps> = ({
...
+		<div>
+			{type === 'textarea' ? (
+				<textarea {...sharedProps} />
+			) : (
+				<input type={inputType} {...sharedProps} />
+			)}
+			{error && (
+				<span className={errorMessageStyles}>{error.message}</span>
+			)}
+		</div>
```
Comment (malyshkoserhii) [inline packages/frontend/src/shared/components/textinput/textinput.component.tsx:66]: split on two components: input and textarea

**Pull request merged**